src/panels/lovelace/elements/hui-conditional-element.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  Condition,
+  checkConditionsMet,
+  validateConditionalConfig,
+} from "../../lovelace/common/validate-condition";
+import { createConfiguredHuiElement } from "../../lovelace/cards/picture-elements/create-configured-hui-element";
+
+import { LovelaceElement, LovelaceElementConfig } from "./types";
+import { HomeAssistant } from "../../../types";
+
+interface Config extends LovelaceElementConfig {
+  conditions: Condition[];
+  elements: LovelaceElementConfig[];
+}
+
+class HuiConditionalElement extends HTMLElement implements LovelaceElement {
+  public _hass?: HomeAssistant;
+  private _config?: Config;
+  private _elements: LovelaceElement[] = [];
+
+  public setConfig(config: Config): void {
+    if (
+      !config.conditions ||
+      !Array.isArray(config.conditions) ||
+      !config.elements ||
+      !Array.isArray(config.elements) ||
+      !validateConditionalConfig(config.conditions)
+    ) {
+      throw new Error("Error in card configuration.");
+    }
+
+    this._config = config;
+
+    this._config.elements.map((elementConfig: LovelaceElementConfig) => {
+      this._elements.push(
+        createConfiguredHuiElement(elementConfig, this._hass!)
```

Comment:
You don't have `this._hass` here. We always call `setConfig` first.

Reply:
`createConfiguredHuiElement` should not take `hass` as that's not needed for creation of the element.

Reply:
It does need hass to provide it to the element. This is exactly what the local _createHuiElement method used to do and it is required (not for creation but for full operation).